sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -678,6 +678,9 @@ static String getExtractDestinationUri(String extractDestinationDir) {
     // The maximum number of retry jobs.
     static final int MAX_RETRY_JOBS = 3;
 
+    // The maximum size of a single file - 14TiB, just under the 15 TiB limit.
+    static final long MAX_FILE_SIZE = 14 * (1L << 40);
```

Comment:
good catch - this was a typo! 

To be honest, I think giving 1 TB of slack is a bit excessive. We could probably stay within 1gb of the limits, and still be perfectly safe